src/predictors/gtfs-position-tracker.cc

```diff
@@ -22,67 +22,82 @@ void GTFSPositionTracker::predict(
+          transit_realtime::TripUpdate* tripUpdateToUpdate;
+          bool stopTimeUpdateExists = false;
+          transit_realtime::TripUpdate_StopTimeEvent* depatureToUpdate;
+
+          for (int i = 0; i < outputFeed.entity_size(); ++i) {
+            const transit_realtime::FeedEntity& outputFeedEntity = outputFeed.entity(i);
+            if (outputFeedEntity.has_trip_update() && outputFeedEntity.trip_update().trip().trip_id() == tripID) {
+              tripUpdateExists = true;
+              tripUpdateToUpdate = outputFeed.mutable_entity(i)->mutable_trip_update();
+                for (int j = 0; j < outputFeedEntity.trip_update().stop_time_update_size(); ++j) {
+                const transit_realtime::TripUpdate_StopTimeUpdate& update = outputFeedEntity.trip_update().stop_time_update(j);
+                if (update.stop_id() == location.id_) {
+                  stopTimeUpdateExists = true;
+                  depatureToUpdate = tripUpdateToUpdate->mutable_stop_time_update(j)->mutable_departure();
+                  break;
                 }
               }
             }
+          }
 
-            auto now = std::chrono::system_clock::now();
-            auto current_time = std::chrono::duration_cast<std::chrono::seconds>(now.time_since_epoch()).count();
-            
-            if (tripUpdateExists) {
-              stopTimeUpdate.mutable_departure()->set_time(std::max(current_time, stopTimeUpdate.departure().time()));
-            } else {
-              transit_realtime::FeedEntity* new_entity = outputFeed.add_entity();
-              new_entity->set_id(tripID);
-
-              transit_realtime::TripUpdate* trip_update = new_entity->mutable_trip_update();
-              trip_update->mutable_trip()->set_trip_id(tripID);
-
-              transit_realtime::TripUpdate_StopTimeUpdate* stop_time_update = trip_update->add_stop_time_update();
-              stop_time_update->set_stop_id(location.id_);
-              stop_time_update->mutable_departure()->set_time(current_time);
+          auto now = std::chrono::system_clock::now();
+          auto current_time = std::chrono::duration_cast<std::chrono::seconds>(now.time_since_epoch()).count();
+          
+          if (!tripUpdateExists){
+            transit_realtime::FeedEntity* new_entity = outputFeed.add_entity();
+            new_entity->set_id(tripID);
 
-              std::cerr << "Create trip update" << timetable.locations_.ids_.size() << std::endl;
-            }
+            tripUpdateToUpdate = new_entity->mutable_trip_update();
+            transit_realtime::TripDescriptor* trip = tripUpdateToUpdate->mutable_trip();
+            trip->set_trip_id(tripID);
+            trip->set_route_id(routeID);
+            tripUpdateToUpdate->mutable_vehicle()->set_id(vehicleID);
+          }
 
+          if (!stopTimeUpdateExists){
+            transit_realtime::TripUpdate_StopTimeUpdate* stop_time_update = tripUpdateToUpdate->add_stop_time_update();
```

Comment:
tripUpdate might not be initialized here. Maybe we should check the path where this could happen and handle that case explicitly.